src/app/folder/folderlist.component.html

```diff
@@ -5,15 +5,19 @@
   </button>
 </div>
 <mat-tree [dataSource]="dataSource" [treeControl]="treeControl">
-    <mat-tree-node *matTreeNodeDef="let node" matTreeNodePadding
-      matTreeNodePaddingIndent="20" style="display: flex"
+    <mat-tree-node *matTreeNodeDef="let node" [matTreeNodePadding]="node.folderLevel"
+      matTreeNodePaddingIndent="20"
+      style="display: flex"
+      class="mailFolder"
       [ngClass]="{
         'selectedFolder' : selectedFolder == node.folderPath,
-        'dropToFolderAllowed' : dropFolderId === node.folderId
+        'dropToFolderAllowed' : dropFolderId===node.folderId && dropAboveOrBelowOrInside === 3,
```

Comment:
The checks don't seem to be using enum names – is there a reason to not use them in HTML?

Reply:
No - it's more readable using them in HTML too, so I just had to find out how. I used the method described here: https://stackoverflow.com/questions/35835984/how-to-use-a-typescript-enum-value-in-an-angular2-ngswitch-statement/35835985#35835985

Looks better now?

Reply:
Lovely!